lib/trino-parquet/src/main/java/io/trino/parquet/reader/Int64TimestampNanosColumnReader.java

```diff
@@ -55,6 +62,22 @@ else if (type == TIMESTAMP_NANOS) {
                     floorDiv(epochNanos, NANOSECONDS_PER_MICROSECOND),
                     floorMod(epochNanos, NANOSECONDS_PER_MICROSECOND) * PICOSECONDS_PER_NANOSECOND));
         }
+        else if (type == TIMESTAMP_TZ_MILLIS) {
+            type.writeLong(blockBuilder, DateTimeEncoding.packDateTimeWithZone(Timestamps.round(epochNanos, 6) / NANOSECONDS_PER_MILLISECOND, 0));
+        }
+        else if (type == TIMESTAMP_TZ_MICROS) {
+            long roundedEpochNanos = Timestamps.round(epochNanos, 3);
+            type.writeObject(blockBuilder, LongTimestampWithTimeZone.fromEpochMillisAndFraction(
+                    floorDiv(roundedEpochNanos, NANOSECONDS_PER_MILLISECOND),
+                    floorMod(roundedEpochNanos, NANOSECONDS_PER_MILLISECOND) * PICOSECONDS_PER_NANOSECOND,
+                    UTC_KEY));
+        }
+        else if (type == TIMESTAMP_TZ_NANOS) {
+            type.writeObject(blockBuilder, LongTimestampWithTimeZone.fromEpochMillisAndFraction(
+                    floorDiv(epochNanos, NANOSECONDS_PER_MILLISECOND),
+                    floorMod(epochNanos, NANOSECONDS_PER_MILLISECOND) * PICOSECONDS_PER_NANOSECOND,
+                    UTC_KEY));
```

Comment:
Are we assuming that the writer has converted timestamps to UTC before writing ?
Should we be verifying that `TimestampLogicalTypeAnnotation#isAdjustedToUTC` is true ?

Reply:
It might make sense to look at `isAdjustedToUTC`, but this is not how it's implemented for millis/micros precision (e.g.
[Int64TimestampMillisColumnReader](https://github.com/trinodb/trino/blob/05a0efa9232b0e13505f0938ecce9cd845f1b02a/lib/trino-parquet/src/main/java/io/trino/parquet/reader/Int64TimestampMillisColumnReader.java) ). Changing this is likely out of scope of this PR, as it would require thinking about backward compatibility.